storage: test failure in  TestRefreshPendingCommands

Found in:
https://teamcity.cockroachdb.com/viewLog.html?buildId=80840&buildTypeId=Cockroach_UnitTests

```
I161205 17:23:22.493571 12689 storage/replica_raftstorage.go:639 [s3,r1/3:/M{in-ax},@c420b80480] applied Raft snapshot in 0ms [clear=0ms batch=0ms entries=0ms commit=0ms] 
I161205 17:23:22.591587 12680 storage/replica.go:2269 [s3,r1/3:/M{in-ax},@c420b80480] not quiescing: 1 pending commands 
I161205 17:23:22.741435 12667 storage/replica.go:2269 [s3,r1/3:/M{in-ax},@c420b80480] not quiescing: 1 pending commands 
I161205 17:23:22.791485 12670 storage/replica.go:2269 [s3,r1/3:/M{in-ax},@c420b80480] not quiescing: 1 pending commands 
...
I161205 17:25:16.341576 12677 storage/replica.go:2269 [s3,r1/3:/M{in-ax},@c420b80480] not quiescing: 1 pending commands 
I161205 17:25:16.391584 12668 storage/replica.go:2269 [s3,r1/3:/M{in-ax},@c420b80480] not quiescing: 1 pending commands 
panic: test timed out after 2m0s 
goroutine 12795 [running]: panic(0x1808960, 0xc420c4f2a0) 
/usr/local/go/src/runtime/panic.go:500 +0x1a1 
testing.startAlarm.func1() 
/usr/local/go/src/testing/testing.go:918 +0x10b 
created by time.goFunc 
/usr/local/go/src/time/sleep.go:154 +0x44
```

I noticed #11771 too but it seems to be a different failure.

My failure to reproduce was from Friday. I tried again just now and the test died fairly quickly:

```
~ make stress TAGS=deadlock PKG=./storage TESTS=TestRefreshPendingCommands STRESSFLAGS="-maxfails 1 -stderr -p 16"
...
364 runs completed, 1 failures, over 1m23s
```

The failure was different from a timed out test:

```
    --- FAIL: TestRefreshPendingCommands/#02 (9.62s)
    	client_raft_test.go:978: store 1: lease repl={2 2 2} start=0.900000123,17 exp=4.500000127,11 pro=3.600000127,12 is still active
```

@spencerkimball Is this related to the epoch-based range leases change?

--- a/pkg/storage/client_raft_test.go
+++ b/pkg/storage/client_raft_test.go
@@ -910,9 +910,14 @@ func TestRefreshPendingCommands(t *testing.T) {
 	// insufficient on its own. In addition, there is a fourth reproposal
 	// mechanism (reasonNewLeaderOrConfigChange) which is not relevant to this
 	// scenario.
+	//
+	// We don't test with only reasonNewLeader because that mechanism is less
+	// robust than refreshing due to snapshot or ticks. In particular, it is
+	// possible for node 3 to propose the RequestLease command and have that
+	// command executed by the other nodes but to never see the execution locally
+	// because it is caught up by applying a snapshot.
 	testCases := []storage.StoreTestingKnobs{
 		{DisableRefreshReasonNewLeader: true, DisableRefreshReasonTicks: true},
-		{DisableRefreshReasonSnapshotApplied: true, DisableRefreshReasonTicks: true},
 		{DisableRefreshReasonNewLeader: true, DisableRefreshReasonSnapshotApplied: true},
 	}
 	for _, c := range testCases {

--- a/pkg/storage/replica.go
+++ b/pkg/storage/replica.go
@@ -2369,7 +2369,7 @@ func defaultSubmitProposalLocked(r *Replica, p *ProposalData) error {
 
 	return r.withRaftGroupLocked(true, func(raftGroup *raft.RawNode) (bool, error) {
 		if log.V(4) {
-			log.Infof(ctx, "proposing command %x", p.idKey)
+			log.Infof(ctx, "proposing command %x: %s", p.idKey, p.Request.Summary())
 		}
 		// We're proposing a command so there is no need to wake the leader if we
 		// were quiesced.

